src/Microsoft.TemplateEngine.Cli/TemplateListCoordinator.cs

```diff
@@ -71,22 +71,32 @@ internal class TemplateListCoordinator
             }
             else
             {
-                // No templates found matching the following input parameter(s): {0}.
-                Reporter.Error.WriteLine(
-                    string.Format(
-                        LocalizableStrings.NoTemplatesMatchingInputParameters,
-                        GetInputParametersString(args/*, appliedParameterMatches*/))
-                    .Bold().Red());
-
-                if (resolutionResult.HasTemplateGroupMatches)
+                //if there is no criteria and filters it means that dotnet new list was run but there is no templates installed.
+                if (args.ListNameCriteria == null && !args.AppliedFilters.Any())
+                {
+                    //No templates installed.
+                    Reporter.Error.WriteLine(LocalizableStrings.NoTemplatesFound.Bold().Red());
+                }
+                else
                 {
-                    // {0} template(s) partially matched, but failed on {1}.
+                    // at least one criteria was specified.
+                    // No templates found matching the following input parameter(s): {0}.
                     Reporter.Error.WriteLine(
                         string.Format(
-                            LocalizableStrings.TemplatesNotValidGivenTheSpecifiedFilter,
-                            resolutionResult.TemplateGroups.Count(),
-                            GetPartialMatchReason(resolutionResult, args/*, appliedParameterMatches*/))
+                            LocalizableStrings.NoTemplatesMatchingInputParameters,
+                            GetInputParametersString(args/*, appliedParameterMatches*/))
```

Comment:
Was this commented section forgotten to be removed?

Reply:
no, I decided to keep it internationally. It would be helpful if we do https://github.com/dotnet/templating/issues/4061 eventually. If decision is not to do it, then it can be removed.